Px.Search.Lucene/LuceneSearcher.cs

```diff
@@ -1,36 +1,124 @@
-        public LuceneSearcher(string indexDirectory)
+        public LuceneSearcher(string indexDirectory, string language)
         {
             if (string.IsNullOrWhiteSpace(indexDirectory))
             {
                 throw new ArgumentNullException("Index directory not defined for Lucene");
             }
+          
+            FSDirectory fsDir = FSDirectory.Open(Path.Combine(indexDirectory, language));
 
-            _indexDirectory = indexDirectory;
+            IndexReader reader = DirectoryReader.Open(fsDir);
+            _indexSearcher = new IndexSearcher(reader);
         }
-
-        public IEnumerable<SearchResult> Find(string searchExpression, string language)
+        /// <summary>
+        /// Search the right index depending on the language and give back a search result
+        /// </summary>
+        /// <param name="searchExpression"></param>
+        /// <param name="language"></param>
+        /// <param name="pageSize"></param>
+        /// <param name="pageNumber"></param>
+        /// <returns></returns>
+        public IEnumerable<SearchResult> Find(string searchExpression, string language, int pageSize, int pageNumber)
         {
-            //Search the right index depending on the language and give back a search result.
             // See https://github.com/statisticssweden/Px.Search.Lucene/blob/main/Px.Search.Lucene/LuceneSearcher.cs
-            throw new NotImplementedException();
+
+            var skipRecords = pageSize * (pageNumber - 1);
+            List<SearchResult> searchResultList = new List<SearchResult>();
+            string[] fields = GetSearchFields();
+            LuceneVersion luceneVersion = LuceneVersion.LUCENE_48;
+
+            QueryParser qp = new MultiFieldQueryParser(luceneVersion,
+                                                       fields,
+                                                       new StandardAnalyzer(luceneVersion));
+            qp.DefaultOperator = _defaultOperator;
+            Query q = qp.Parse(searchExpression);
+            TopDocs topDocs = _indexSearcher.Search(q,skipRecords+pageSize);
+            ScoreDoc[] scoreDocs = topDocs.ScoreDocs;
+
+            for (int i = skipRecords; i < topDocs.TotalHits; i++)
+            {
+                if (i > (skipRecords + pageSize) - 1)
+                {
+                    break;
+                }
+                Document doc = _indexSearcher.Doc(scoreDocs[i].Doc);
+                var searchResult = new SearchResult(
+                    doc.Get(SearchConstants.SEARCH_FIELD_DOCID),
+                    doc.Get(SearchConstants.SEARCH_FIELD_TITLE),
+                    doc.Get(SearchConstants.SEARCH_FIELD_CATEGORY),
+                    doc.Get(SearchConstants.SEARCH_FIELD_FIRSTPERIOD),
+                    doc.Get(SearchConstants.SEARCH_FIELD_LASTPERIOD),
+                    doc.Get(SearchConstants.SEARCH_FIELD_VARIABLES).Split(" ")
+                );
```

Comment:
We should also set the other properties: Description, Sortcode, Updated and so on...